src/mock_request_rate_worker.h

```diff
@@ -46,12 +46,13 @@ class NaggyMockRequestRateWorker : public RequestRateWorker {
       std::chrono::steady_clock::time_point& start_time,
       const bool serial_sequences,
       const std::shared_ptr<IInferDataManager>& infer_data_manager,
-      std::shared_ptr<SequenceManager> sequence_manager)
+      std::shared_ptr<SequenceManager> sequence_manager, size_t dataset_offset)
       : RequestRateWorker(
             id, thread_stat, thread_config, parser, data_loader, factory,
             on_sequence_model, async, max_threads, using_json_data, streaming,
             batch_size, wake_signal, wake_mutex, execute, start_time,
-            serial_sequences, infer_data_manager, sequence_manager)
+            serial_sequences, infer_data_manager, sequence_manager,
+            dataset_offset)
```

Comment:
As mentioned [here](https://github.com/triton-inference-server/perf_analyzer/pull/366/files#r2058826068), we need to pass the dataset offset down from CustomRequestScheduleManager down to InferContext so they can pick up in the payload dataset where the warmup left off.